nexus/src/app/sagas/loopback_address_create.rs

```diff
@@ -0,0 +1,184 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use super::{NexusActionContext, NEXUS_DPD_TAG};
+use crate::app::sagas::{
+    declare_saga_actions, ActionRegistry, NexusSaga, SagaInitError,
+};
+use crate::authn;
+use crate::authz;
+use crate::db::model::LoopbackAddress;
+use crate::external_api::params;
+use anyhow::Error;
+use dpd_client::types::{Ipv4Entry, Ipv6Entry};
+use serde::{Deserialize, Serialize};
+use std::net::IpAddr;
+use std::sync::Arc;
+use steno::ActionError;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Params {
+    pub serialized_authn: authn::saga::Serialized,
+    pub loopback_address: params::LoopbackAddressCreate,
+}
+
+declare_saga_actions! {
+    loopback_address_create;
+    CREATE_LOOPBACK_ADDRESS_RECORD -> "created_loopback_address_record" {
+        + slc_loopback_address_create_record
+        - slc_loopback_address_delete_record
+    }
+    CREATE_LOOPBACK_ADDRESS -> "create_loopback_address" {
+        + slc_loopback_address_create
+    }
+}
+
+#[derive(Debug)]
+pub struct SagaLoopbackAddressCreate;
+impl NexusSaga for SagaLoopbackAddressCreate {
+    const NAME: &'static str = "loopback-address-create";
+    type Params = Params;
+
+    fn register_actions(registry: &mut ActionRegistry) {
+        loopback_address_create_register_actions(registry);
+    }
+
+    fn make_saga_dag(
+        _params: &Self::Params,
+        mut builder: steno::DagBuilder,
+    ) -> Result<steno::Dag, SagaInitError> {
+        builder.append(create_loopback_address_record_action());
+        builder.append(create_loopback_address_action());
+
+        Ok(builder.build()?)
+    }
+}
+
+async fn slc_loopback_address_create_record(
+    sagactx: NexusActionContext,
+) -> Result<LoopbackAddress, ActionError> {
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params::<Params>()?;
+    let nexus = osagactx.nexus();
+
+    let opctx = crate::context::op_context_for_saga_action(
+        &sagactx,
+        &params.serialized_authn,
+    );
+
+    let address_lot_lookup = nexus
+        .address_lot_lookup(&opctx, params.loopback_address.address_lot.clone())
+        .map_err(ActionError::action_failed)?;
+    let (.., authz_address_lot) = address_lot_lookup
+        .lookup_for(authz::Action::Modify)
```

Comment:
I'm not sure of the parent-child relationship between the lot and the address, but we may want this to be `Action::CreateChild`.